stdlib/unittest/mock.pyi

```diff
@@ -234,7 +235,7 @@ class _patch(Generic[_T]):
     @overload
     def __call__(self, func: _TT) -> _TT: ...
```

Comment:
I don't think we need this overload anymore as types should be callable. But can't hurt to add another test for that.

Reply:
I think we still do, so that you can resolve attributes of the class. Note that this only matters when using pyright, since mypy sort of ignores class decorators. I'll add a test case though!